src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -210,15 +210,33 @@ boolean existsMediaFileIn2StorageLocations(MediaEntity media,
             AND eod2.status = :status2
             AND eod.externalLocationType = :location1
             AND eod2.externalLocationType = :location2
-            AND eod.lastModifiedDateTime < :lastModifiedBefore
+            AND eod2.lastModifiedDateTime <= :lastModifiedBefore
```

Comment:
@cmagowan-hmcts correct me if i'm wrong but my ticket seems to suggest:-

1) the media record is the record against which the date check should be done
2) the date check should include an equals

Reply:
i wouldn't change the logic of this method, as that will have repercussions on the methods already using it.
would you not just have location1 as `ARM` and location2 as `incoming` so that this method does not change?

1. I think it should be against how long it has been in ARM, only if it has been in ARM for 24hrs, should we then delete the incoming file. If ARM was down for a week, then the media record could have been sitting there for a week, and not migrated to ARM, so we should not be using the media record for anything.

2. i don't think this will matter...we are talking about nanosecond precision! its unlikely there will be a file that was transferred to ARM at that specific point, and if there was, we woun;t care too much, it will just get marked for deletion the next time the job runs.

Reply:
just thinking about this, i can see that your above change may be required